src/main/java/by/exadel/internship/config/jwt/JwtService.java

```diff
@@ -0,0 +1,60 @@
+package by.exadel.internship.config.jwt;
+
+
+import by.exadel.internship.entity.UserDetailsImpl;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.MalformedJwtException;
+import io.jsonwebtoken.SignatureAlgorithm;
+import lombok.extern.slf4j.Slf4j;
+import org.slf4j.MDC;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Component;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Date;
+
+@Component
+@Slf4j
+public class JwtService {
+    @Value("${key.jwtSecret}")
+    private String jwtSecret;
+
+    @Value("${key.jwtExpirationSec}")
+    private int jwtExpirationSec;
+
+    public String generateJwtToken(Authentication authentication) {
+
+        UserDetailsImpl userPrincipal = (UserDetailsImpl) authentication.getPrincipal();
+        Instant issuedAt = Instant.now().truncatedTo(ChronoUnit.SECONDS);
+        Instant expiration = issuedAt.plus(jwtExpirationSec, ChronoUnit.SECONDS);
+        return Jwts.builder()
+                .setSubject((userPrincipal.getUsername()))
+                .setIssuedAt(Date.from(issuedAt))
+                .setExpiration(Date.from(expiration))
+                .signWith(SignatureAlgorithm.HS512, jwtSecret)
+                .compact();
+    }
+
+    public boolean validateJwtToken(String jwt) {
+        try {
+            Jwts.parser()
+                    .setSigningKey(jwtSecret)
+                    .parseClaimsJws(jwt);
```

Comment:
Is it full validation? How about checking token expiration date?

Reply:
Method parser inside check expiration date, I can't find it inside, but i tested and it's work

Reply:
Thank you for attentive testing!